### DISCUSSION TITLE
run github actions without setup, locally

### DISCUSSION THREAD

**Discussion by nnWhisperer** [nektos/act#2518] [Q&A]
Hello,

Since github actions provide ways to set up binaries such as [that of go](https://github.com/actions/setup-go) or npm, I want to use them on terminal instead of looking up how to install that binary. I found this act repository but it seems that one may only use those actions if they are written inside a workflow yml file. My question is, can we use act to run an action on the local bash?

**For example something like that which installs go in the terminal it is run on:**
```
act actions/setup-go@v5 --with go-version: '^1.13.1'
```

**Comment by ChristopherHX** [comment]
This is a nice idea, but no act does't offer something like this yet. The node requirement could be a problem for running those exensions...

Currently `-P -self-hosted` would make act to run the setup action on your system.

How would act a binary add something to your system path of your current shell? dotnet tool install tells you that the path it used for dotnet tools is not currently in your path, but setup-actions vary much more at that point.
